client/rpc/rpc.go

```diff
@@ -16,20 +22,62 @@ var (
 )
 
 // MakeMessage handles making tx message
-func MakeMessage(t string, payload interface{}) types.Tx {
+func MakeMessage(t string, nonce uint32, payload interface{}, sign bool) (types.Tx, error) {
```

Comment:
Think about this situation: we got a key to sign a transaction, but we don't care how. we just got a key. then, we just need a function to make a formed or formatted message. one unit of a code does one thing: make something from the prepared ingredients. think about layers or hierarchy of functions.